### ISSUE TITLE
Mapping Timestamp to OffsetDateTime in postgres ignore system timezone, the data offset is always UTC

### ISSUE THREAD

**Issue by fedxyz** [jdbi/jdbi#1933]
Hi,

If I query for a column that is of type `timestamptz` and I map it to `OffsetDateTime ` the zone offset is always UTC, for example:

```
jdbi.useHandle(h -> {
	h.createQuery(query).mapTo(OffsetDateTime.class).list().forEach(System.out::println);
});
```
where query is something like:
`select some_timestamptz_col from table`

this will return for example:
```
2021-10-17T21:22:36.188737Z
2021-10-17T21:22:36.187334Z
2021-10-17T21:22:36.185780Z
2021-10-17T21:22:36.181306Z
```

the data are correct but they are "formatted" in UTC, my system timezone is ignored (it is not UTC) and postgres timezone is not UTC too. 
I was expecting the ZoneOffset to be the the same as my system timezone offset.

The version I am testing is `3.23.0`, I am using PostgresPlugin too.

**Comment by fedxyz** [comment]
testing a bit I found that PostgresPlugin change this behaviour.

Same query/code on a `timestamptz` field:

```
jdbi.useHandle(h -> {
	h.createQuery(query).mapTo(OffsetDateTime.class).list().forEach(System.out::println);
});
```
with PostgresPlugin installed, a sample output is:
```
2021-10-17T19:23:38.707480Z
2021-10-17T19:22:06.645620Z
2021-10-17T19:19:29.750619Z
```

without the PostgresPlugin installed:

```
2021-10-17T21:23:38.707480+02:00
2021-10-17T21:22:06.645620+02:00
2021-10-17T21:19:29.750619+02:00
```
That I consider consistent with my system timezone.

Why this different behaviour?

Thanks for the help.

**Comment by hgschmie** [comment]
What the postgres plugin does is delegate the conversion of timestamps to the postgres jdbi driver. It registers code that will call `Statement#setObject` and `ResultSet#getObject` with the OffsetDateTime object thus letting the driver convert it from and to the database type. It seems that the driver does not do this zone conversion as in your example.

The equivalent code in pure JDBC would be calling `setObject` and `getObject` as described above. Could you maybe test whether that behaves different than what you expect?

**Comment by qualidafial** [comment]
Ok, I see what you're getting at.

JDBC in JDK 8 only guarantees support for java.time.Instant, and leaves out all the *DateTime types. In order to support them in bare Jdbi, we convert them to Instant where reasonable.

However, since the Postgres JDBC driver supports those types directly, we use that functionality in the Postgres plugin.

**Comment by fedxyz** [comment]
When I wrote the first 2 comments I was not aware of the jdbc documentation that I linked in my third comment:
https://jdbc.postgresql.org/documentation/head/8-date-time.html

As you @hgschmie said this explain the behaviour with postgres plugin that is different with jdbi default one.
For me the issue could be considered closed.

I was just thinking if maybe it makes sense when using postgres plugin the possibility to specify some classes that still use jdbi default mapping, in this case for example `OffsetDateTime`.
Or an alternative mapper that with postgres plugin still rely on jdbc driver conversion but it then do:

`offsetDateTime.atZoneSameInstant(ZoneId.systemDefault()).toOffsetDateTime()`
